core/kernel/tpm.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2020, ARM Limited
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <kernel/generic_boot.h>
+#include <kernel/dt.h>
+#include <mm/core_mmu.h>
+#include <mm/core_memprot.h>
+#ifdef CFG_DT
+#include <kernel/dt.h>
+#include <libfdt.h>
+#endif
+#include <kernel/tpm.h>
+#include <string.h>
+
+static vaddr_t tpm_log_addr;
```

Comment:
This is only used as a `void *` so it would be more convenient to use that type here instead.